### PR TITLE
[FLINK-30008][Connectors/AWS] Upgrade to Flink 1.16.0 and add CI build

### PR DESCRIPTION
**Pull request by dannycranmer**
## What is the purpose of the change

Add support for Flink 1.16

## Brief change log

* Set default Flink version to 1.16.0
* Remove snapshot repo
* Add CI build profile for 1.16.0
* Syncing parent pom to elasticsearch in prep for release
* Adding git submodule for connector release scripts

## Verifying this change

Run locally

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? n/a

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,145 +18,239 @@ specific language governing permissions and limitations
...
+        <url>https://github.com/apache/flink-connector-aws</url>
+        <connection>git@github.com:apache/flink-connector-aws.git</connection>
+        <developerConnection>scm:git:https://gitbox.apache.org/repos/asf/flink-connector-aws.git</developerConnection>
+    </scm>
+
+    <properties>
+        <aws.sdk.version>2.17.247</aws.sdk.version>
+        <netty.version>4.1.77.Final</netty.version>
+        <flink.version>1.16.0</flink.version>
+        <flink.shaded.version>15.0</flink.shaded.version>
```
Comment (hlteoh37) [inline pom.xml:59]: Q. Why is this 15.0? We see this in the outputs

```
Warning:  Could not find artifact org.apache.flink:flink-sql-connector-dynamodb:jar:1.15.0 in central (https://repo.maven.apache.org/maven2)
Warning:  Could not resolve dependency with descriptor 'org.apache.flink:flink-sql-connector-dynamodb:1.15.0'.
Warning:  Please provide at least one resolvable old version using one of the configuration elements <oldVersion/> or <oldVersions/>.

```

**Review by hlteoh37 — COMMENTED** [review]
Had 2 questions, otherwise LGTM

Is there a way we can upgrade the maven checkstyle plugin version?

File: pom.xml (hunk 1 above)
Comment (dannycranmer) [reply pom.xml:59]: I copied this from elasticsearch actually. But I do see a 16.0 available. https://mvnrepository.com/artifact/org.apache.flink/flink-shaded-force-shading

Not sure on the warnings, will take a look.

Comment (dannycranmer) [reply pom.xml:59]: These things are unrelated. I have upgraded Flink shaded to 16.0. The warnings were from the `japicmp` plugin which is trying to compare binary compatibility. Given that this is the first version of the DDB connector there is no baseline to compare against. I have removed this plugin, we can add it back on the next version

**Review by hlteoh37 — APPROVED** [review]
LGTM

**Pull request merged**